backend/src/auth/jwt.guard.ts

```diff
@@ -17,6 +27,38 @@ export class JwtAuthGuard extends AuthGuard("jwt") {
 		if (isPublic) {
 			return true;
 		}
-		return super.canActivate(context);
+
+		try {
+			const request = context.switchToHttp().getRequest();
+			const token = request.headers.authorization?.split(" ")[1];
+			if (!token) {
+				throw new UnauthorizedException("Unauthorized", {
+					cause: new Error(),
+					description: "Token not found",
+				});
+			}
+
+			const secretKey = this.configService.get<string>("JWT_AUTH_SECRET");
+			const decoded = jwt.verify(token, secretKey);
+			request.user = decoded;
+			return super.canActivate(context);
```

Comment:
We use jwt auth login provided from `AuthGuard`.
So, I don't want to implement jwt verify logic in CodePair.

Then, we can get the exception by overriding `handleRequest` in `AuthGuard`.

```typescript
handleRequest(err, user, info, context: ExecutionContext) {
    if (err || !user) {
	console.error("Authentication error:", err);
	console.error("Additional info:", info);

	// We can throw an exception we want
	throw err;
    }

    return user;
}
```

Reply:
I fixed it. Thank you!